tfutils/base.py

```diff
@@ -365,7 +363,7 @@ def run(sess,
         assert (step > old_step), (step, old_step)
         if train_results['loss'] > thres_loss:
             raise HiLossError('Loss {:.2f} exceeded the threshold {:.2f}'.format(train_results['loss'], thres_loss))
-        if step % saver.save_valid_freq == 0 and valid_targets is not None:
```

Comment:
I see. The problem was that on line 570 `valid_targets_dict[vtarg] = vtargets` if this dict is None, the assignment will fail. `test.py` would not catch it because it doesn't have validation targets specified.

BUT: You're very right asking if I ran `test.py`. I didn't and I should have. I'm still not used to proper code development; sorry about these occasional blunders. Perhaps there is a (standard) way to make sure tests are run prior to pushing?

Instead of `test.py` I was running `tconvnet` that had validation targets always defined.